### PR TITLE
Add --gzip-requests to compress request bodies in feed client

### PR DESCRIPTION
**Pull request by jonmv**
@bjorncs please review. I verified this works with jetty on our factory app.

### CODE REVIEW

File: vespa-feed-client-cli/src/main/java/ai/vespa/feed/client/impl/CliArguments.java [hunk 1]
```diff
@@ -58,6 +58,7 @@ class CliArguments {
     private static final String STDIN_OPTION = "stdin";
     private static final String DOOM_OPTION = "max-failure-seconds";
     private static final String PROXY_OPTION = "proxy";
+    private static final String GZIP_REQUESTS_OPTION = "gzip-requests";
```
Comment (baldersheim) [inline vespa-feed-client-cli/src/main/java/ai/vespa/feed/client/impl/CliArguments.java:61]: I think --compression=gzip is better.
Then it is very easy to support other compression schemes later on.

Comment (jonmv) [reply vespa-feed-client-cli/src/main/java/ai/vespa/feed/client/impl/CliArguments.java:61]: That's a very good suggestion. I'll change it.

File: vespa-feed-client/pom.xml [hunk 2]
```diff
@@ -46,6 +46,11 @@
       <artifactId>junit-jupiter</artifactId>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>com.github.tomakehurst</groupId>
+      <artifactId>wiremock-jre8-standalone</artifactId>
+      <scope>test</scope>
+    </dependency>
```
Comment (baldersheim) [inline vespa-feed-client/pom.xml:53]: I think this one was not intended either.

Comment (jonmv) [reply vespa-feed-client/pom.xml:53]: It was, but I forgot to add the test.

**Review by baldersheim — APPROVED** [review]

File: vespa-feed-client/src/test/java/ai/vespa/feed/client/impl/ApacheClusterTest.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
+package ai.vespa.feed.client.impl;
```
Comment (bjorncs) [inline vespa-feed-client/src/test/java/ai/vespa/feed/client/impl/ApacheClusterTest.java:1]: Missing copyright header

File: vespa-feed-client/src/test/java/ai/vespa/feed/client/impl/ApacheClusterTest.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+import static com.github.tomakehurst.wiremock.client.WireMock.anyRequestedFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.anyUrl;
+import static com.github.tomakehurst.wiremock.client.WireMock.equalTo;
+import static com.github.tomakehurst.wiremock.client.WireMock.okJson;
+import static com.github.tomakehurst.wiremock.client.WireMock.postRequestedFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class ApacheClusterTest {
```
Comment (bjorncs) [inline vespa-feed-client/src/test/java/ai/vespa/feed/client/impl/ApacheClusterTest.java:32]: Consider adding author

**Review by bjorncs — COMMENTED** [review]
We might want to have gzip compression being enabled by default since it's already supported by all versions of jdisc container.

**Pull request merged**